src/test_trie.py

```diff
@@ -0,0 +1,137 @@
+# _*_ encoding: utf-8 _*_
+import pytest
+
+
+@pytest.fixture()
+def full_trie():
+    from trie import Trie
+    new_trie = Trie()
+    new_trie.insert('words')
+    new_trie.insert('trie')
+    new_trie.insert('trip')
+    new_trie.insert('in')
+    new_trie.insert('i')
+    new_trie.insert('worldly')
+    return new_trie
+
+
+def test_trie():
+    """Test that we can insert into the Trie."""
+    from trie import Trie
+    new_trie = Trie()
+    new_trie.insert('words')
+    assert new_trie.root == {'w': {'o': {'r': {'d': {'s': {'$': '$'}}}}}}
+
+
+def test_2_trie():
+    """Test that we can insert two words into the Trie."""
+    from trie import Trie
+    new_trie = Trie()
+    new_trie.insert('words')
+    new_trie.insert('trie')
+    assert new_trie.root == {'w': {'o': {'r': {'d': {'s': {'$': '$'}}}}},
+                             't': {'r': {'i': {'e': {'$': '$'}}}}}
+
+
+def test_overlapping_words():
+    from trie import Trie
+    new_trie = Trie()
+    new_trie.insert('words')
+    new_trie.insert('trie')
+    new_trie.insert('trip')
+    assert new_trie.root == {'w': {'o': {'r': {'d': {'s': {'$': '$'}}}}},
```

Comment:
The problem with this is....
Dictionaries are by nature non-stable. By testing this way, you are relying on the stability of the dict you have hard coded here.